sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -488,6 +489,16 @@ object SimplifyBinaryComparison
         // False with inequality
         case a GreaterThan b if canSimplifyComparison(a, b, notNullExpressions) => FalseLiteral
         case a LessThan b if canSimplifyComparison(a, b, notNullExpressions) => FalseLiteral
+
+        // One side is Literal
+        case (a: Predicate) EqualTo TrueLiteral => a
```

Comment:
Does `a` have to be a `Predicate`? It seems like we can apply this optimization to any expression.

Reply:
I guess the idea is for push down benefit. But for other expressions, it should be also okay.

Reply:
Thanks, yes the intention is for pushing down the filter. I will add a comment.
Actually, my original attempt, which I later changed to `Predicate` only, was `case a EqualTo TrueLiteral if a.dataType.isInstanceOf[BooleanType]`. I found that is too disruptive for some tests. I will keep it only for `Predicate` for now, and we can still get the push down benefit.

Reply:
updated